packages/api/src/services/subscriptions.ts

```diff
@@ -56,28 +81,22 @@ export const unsubscribe = async (
   return getRepository(Subscription).save(subscription)
 }
 
-const sendUnsubscribeEmail = async (
-  unsubscribeMailTo: string,
+export const unsubscribeAll = async (
+  userId: string,
   newsletterEmail: string
 ): Promise<void> => {
-  const sent = await sendEmail({
-    to: unsubscribeMailTo,
-    subject: 'Unsubscribe',
-    text: `This message was automatically generated by Omnivore.`,
-    from: newsletterEmail,
-  })
-
-  if (!sent) {
-    throw new Error(`Failed to unsubscribe, email: ${unsubscribeMailTo}`)
-  }
-}
+  try {
+    const subscriptions = await getRepository(Subscription).find({
+      where: {
+        user: { id: userId },
+        newsletterEmail,
+      },
+    })
 
-const sendUnsubscribeHttpRequest = async (url: string): Promise<void> => {
-  const response = await fetch(url, {
-    method: 'GET',
-  })
-
-  if (!response.ok) {
-    throw new Error(`Failed to unsubscribe, response: ${response.statusText}`)
+    for (const subscription of subscriptions) {
+      await unsubscribe(subscription)
```

Comment:
One small issue I see, is if one fails to unsubscribe and throws an error, then we end this loop and don't attempt the others.

Reply:
Nice to catch the bug